common/persistence/execution_manager.go

```diff
@@ -937,6 +977,40 @@ func (m *executionManagerImpl) Close() {
 	m.persistence.Close()
 }
 
+func (m *executionManagerImpl) trimHistoryNode(
+	shardID int32,
+	namespaceID string,
+	workflowID string,
+	runID string,
+) {
+	response, err := m.GetWorkflowExecution(&GetWorkflowExecutionRequest{
+		ShardID:     shardID,
+		NamespaceID: namespaceID,
+		WorkflowID:  workflowID,
+		RunID:       runID,
+	})
+	if err != nil {
```

Comment:
I understand why this only needs to be best-efforts, but still it should be VERY rare that GetWorkflowExecution fails here.  Did you consider logging a warning here?  If this lookup fails it seems likely it's a bug somewhere.

Maybe (eventually) it's even a candidate for a panic in dev? ... not suggesting we do that now.

Reply:
>Maybe (eventually) it's even a candidate for a panic in dev?

this error can happen if DB is temporarily unavailable, so error can be expected (compared to in mem state check)

added logging here, with namespace & workflow & run.